# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium-shim/patches/with_update/revert_TargetNtSetInformationThread_change.patch
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium-shim/patches/with_update/revert_TargetNtSetInformationThread_change.patch
--- a/security/sandbox/chromium-shim/patches/with_update/revert_TargetNtSetInformationThread_change.patch	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium-shim/patches/with_update/revert_TargetNtSetInformationThread_change.patch	Wed Jun 12 11:10:48 2019 +0100
@@ -9,9 +9,9 @@
 diff --git a/security/sandbox/chromium/sandbox/win/src/policy_target.cc b/security/sandbox/chromium/sandbox/win/src/policy_target.cc
 --- a/security/sandbox/chromium/sandbox/win/src/policy_target.cc
 +++ b/security/sandbox/chromium/sandbox/win/src/policy_target.cc
-@@ -76,16 +76,26 @@ NTSTATUS WINAPI TargetNtSetInformationTh
-     NtSetInformationThreadFunction orig_SetInformationThread, HANDLE thread,
-     NT_THREAD_INFORMATION_CLASS thread_info_class, PVOID thread_information,
+@@ -78,16 +78,26 @@ NTSTATUS WINAPI TargetNtSetInformationTh
+     NT_THREAD_INFORMATION_CLASS thread_info_class,
+     PVOID thread_information,
      ULONG thread_information_bytes) {
    do {
      if (SandboxFactory::GetTargetServices()->GetState()->RevertedToSelf())
@@ -32,7 +32,8 @@
      return STATUS_SUCCESS;
    } while (false);
  
-   return orig_SetInformationThread(thread, thread_info_class,
-                                    thread_information,
-                                    thread_information_bytes);
+   return orig_SetInformationThread(
+       thread, thread_info_class, thread_information, thread_information_bytes);
  }
+ 
+ // Hooks NtOpenThreadToken to force the open_as_self parameter to be set to